### PR TITLE
ARTEMIS-995 Bulk of test fixes

### PR DESCRIPTION
**Pull request by dudaerich**
This commit fixes tests:
 ActiveMQScheduledComponentTest.testAccumulationOwnPool
 PendingDeliveriesTest.testWithtReconnect
 ReceiveTest.testReceiveImmediate
 ActiveMQProducerResourceTest.testSendString
 EmbeddedActiveMQResourceTest.testSendString
 MultipleEmbeddedActiveMQResourcesTest.testMultipleServers
 MultipleEmbeddedJMSResourcesTest.testMultipleServers
 ActiveMQDynamicProducerResourceWithoutAddressTest.testSendString
 ActiveMQDynamicProducerResourceWithoutAddressExceptionTest.testSendBytesToDefaultAddress
 ActiveMQDynamicProducerResourceTest.testSendString
 ActiveMQServerControlTest.testTotalMessageCount
 EmbeddedActiveMQResourceCustomConfigurationTest.testCustomConfiguration
 EmbeddedJMSResourceMultipleFileConfigurationTest.testConfiguration
 EmbeddedJMSResourceQueueTest.testPushObjectMessage
 EmbeddedJMSResourceSingleFileConfigurationTest.testConfiguration
 EmbeddedActiveMQResourceFileConfigurationTest.testConfiguredQueue
 EmbeddedJMSResourceTopicTest.testPushObjectMessage
 LargeMessageFailoverTest.testTimeoutOnFailoverTransactionCommit

### CODE REVIEW

**Comment by tabish121** [comment]
Given the number of changes here I'd say this deserves a JIRA.

**Comment by clebertsuconic** [comment]
But this is not changing any users code.  I'm not sure. 


I actually told him to open a Jira of a real change was made.  But since only tests were changed. I'm not so sure.

**Comment by dudaerich** [comment]
I think JIRA is good place to justification of change and for discussion. As Clebert has pointed out these changes are only test fixes. I don't see any added value to open a JIRA :)

**Comment by tabish121** [comment]
Unless you are just fixes a typo in a comment most every change deserves a JIRA, if you feel the need to change the code then there's a reason and that reason should be documented.

**Comment by dudaerich** [comment]
I see. It's no problem. I'll create a JIRA and update the commit message. Thanks.

**Comment by clebertsuconic** [comment]
Create the JIRA as a TASK then.. cleanup testsuite maybe.

**Comment by dudaerich** [comment]
Good idea Clebert. Thanks.

File: artemis-junit/src/test/java/org/apache/activemq/artemis/junit/MultipleEmbeddedActiveMQResourcesTest.java [hunk 1]
```diff
@@ -38,6 +38,11 @@
    static final String ASSERT_RECEIVED_FORMAT = "Message should have been received from %s";
    static final String ASSERT_COUNT_FORMAT = "Unexpected message count in queue %s";
 
+   static {
+      ThreadLeakCheckRule.addKownThread("MemoryPoolMXBean notification dispatcher");
```
Comment (clebertsuconic) [inline artemis-junit/src/test/java/org/apache/activemq/artemis/junit/MultipleEmbeddedActiveMQResourcesTest.java]: Isn't there a way to kill these threads?

Comment (dudaerich) [inline artemis-junit/src/test/java/org/apache/activemq/artemis/junit/MultipleEmbeddedActiveMQResourcesTest.java]: This was already reported on [HORNETQ-1206](https://issues.jboss.org/browse/HORNETQ-1206) and they argued that these threads are not managed by broker itself. They are created by IBM JDK.
I don't think we can kill some threads but I am not sure.

Comment (dudaerich) [inline artemis-junit/src/test/java/org/apache/activemq/artemis/junit/MultipleEmbeddedActiveMQResourcesTest.java]: They are already ignored also in Artemis test suite [here](https://github.com/apache/activemq-artemis/blob/master/artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java#L216).

**Pull request merged**